tools/rdm/TestDefinitions.py

```diff
@@ -651,20 +661,39 @@ def GetFailed(self):
     self.SendGet()
 
 
-class SetDeviceInfo(ResponderTestFixture, DeviceInfoTest):
-  """SET device info."""
+class SetDeviceInfo(ResponderTestFixture,
+                    DeviceInfoTest):
+  """Attempt to SET device info with no data."""
   CATEGORY = TestCategory.ERROR_CONDITIONS
 
   def Test(self):
-    self.AddExpectedResults(TestMixins.UnsupportedSetNacks(self.pid))
-    self.SendRawSet(ROOT_DEVICE, self.pid)
+    # Unknown PID shouldn't be valid for a mandatory PID
+    self.AddExpectedResults(
+      self.NackSetResult(RDMNack.NR_UNSUPPORTED_COMMAND_CLASS)
```

Comment:
The standard doesn't (yet) say anything about the order of validation, so unknown pid is valid here

Imagine the implementation looks up first by CC and then PID, rather than PID then CC.